nnfabrik/templates/transfer/transferred_trained_model.py

```diff
@@ -0,0 +1,310 @@
+        """
+
+        definition = """
+        collapsed_history:                 varchar(64)
+        model_fn:                     varchar(64)
+        model_hash:                   varchar(64)
+        dataset_fn:                   varchar(64)
+        dataset_hash:                 varchar(64)
+        trainer_fn:                   varchar(64)
+        trainer_hash:                 varchar(64)
+        seed:                         int
+        prev_collapsed_history:             varchar(64)
+        """
+
+        @classmethod
+        def add_entry(cls, key):
+            key = {
+                "model_fn": key["model_fn"],
+                "model_hash": key["model_hash"],
+                "dataset_fn": key["dataset_fn"],
+                "dataset_hash": key["dataset_hash"],
+                "trainer_fn": key["trainer_fn"],
+                "trainer_hash": key["trainer_hash"],
+                "seed": key["seed"],
+                "prev_collapsed_history": key["collapsed_history"],
+            }
+            key["collapsed_history"] = make_hash(key)
+            cls.insert1(key, skip_duplicates=True)
+
+        def get_prev_key(self, key):
+            if key["collapsed_history"]:
+                prev_key = self & {"collapsed_history": key["collapsed_history"]}
+                return prev_key.proj(
+                    "model_fn",
+                    "model_hash",
+                    "dataset_fn",
+                    "dataset_hash",
+                    "trainer_fn",
+                    "trainer_hash",
+                    "seed",
+                    collapsed_history="prev_collapsed_history",
+                    next_collapsed_history="collapsed_history",
+                )
+            else:  # no history yet (i.e. transfer step 0)
+                return self.proj() - self  # return something empty
+
+    def _transfer_recipe(self, transfer_step):
+        """
+        Combines multiple transfer recipes and their resitrictions as specified by post_restr attribute.
+        The combination is transfer-step-specific, meaning only the recipes relevant for a specific transfer step would be combined.
+
+        Combining recipes are pretty easy and the user does not need to interact with this method directly. Below is an example:
+        Let us assume you have two recipe tables: TrainerRecipe and ModelRecipe, the you can attach all these recipes to your
+        TransferTrainedModel table as follow:
+
+        ``` Python
+            TransferTrainedModel.transfer_recipe = [TrainerRecipe, ModelRecipe]
+        ```
+
+        The rest (combining the recipes and their restrictions) is taken care of by this method.
+
+        Args:
+            transfer_step (int): table population trasnfer step.
+
+        Returns:
+            string or datajoint AndList: A single or combined restriction of one or multiple recipes, respectively.
+        """
+
+        if isinstance(self.transfer_recipe, list):
```

Comment:
unless you really want to restrict usage to `list`, consider using `Iterable` and `Sized` from `collections` package

Reply:
The problem is that I don't want strings. I added the acceptance of `tuple`